oauth_application.go

```diff
@@ -0,0 +1,30 @@
+package clerk
+
+type OAuthApplication struct {
+	APIResource
+	Object        string `json:"object"`
+	ID            string `json:"id"`
+	InstanceID    string `json:"instance_id"`
+	Name          string `json:"name"`
+	ClientID      string `json:"client_id"`
+	Public        bool   `json:"public"`
+	Scopes        string `json:"scopes"`
+	CallbackURL   string `json:"callback_url"`
+	AuthorizeURL  string `json:"authorize_url"`
+	TokenFetchURL string `json:"token_fetch_url"`
+	UserInfoURL   string `json:"user_info_url"`
+	CreatedAt     int64  `json:"created_at"`
+	UpdatedAt     int64  `json:"updated_at"`
+}
+
+type OAuthApplicationWithClientSecret struct {
```

Comment:
:question: Do we need an additional type for just one field? How about we add a `*string` field to the `OAuthApplication` type instead?

```go
type OAuthApplication struct {
  ClientSecret *string `json:"client_secret,omitempty"`
}
```

Reply:
You are right this a better approach, I will update the code!